gps-tutorials/esp32tracker/esp32tracker.ino

```diff
@@ -33,8 +37,8 @@ void setup()
   notecard.begin();
 
   // Query ProductUID. If the ProductUID does not match, perform a card.restore
-  if (productUIDMatch()) {
-    serialDebug.println("ProductUID matches, skipping reset...");
+  if (!productUID[0] || productUIDMatch()) {
+    serialDebug.println("ProductUID matches or is not set, skipping reset...");
```

Comment:
I don't see how this sketch can work with a valid `productUID`.

It seems like it needs it's own case and a log indicating that `productUID` must be set.
```suggestion
  if (!productUID[0]) {
    serialDebug.println("ProductUID is not set, sketch cannot proceed...");
    while (true);
  } else if (productUIDMatch()) {
    serialDebug.println("ProductUID matches, skipping reset...");
```

Reply:
Right, I wasn't sure what to do here (and I'd missed the fact that the sketch doesn't run if the product ID hasn't been set on the notecard.) 

I'd like all our example sketches to operate in the same way regarding PRODUCT_UID so that there's no surprises.  At present, most sketches still run even if there's no product UID set.  Should we adopt this approach and not run the sketch if `PRODUCT_UID` isn't defined in the code or on the Notecard? (Accompanied with the compile-time message and details in README.md.)